esp/esp/users/models/__init__.py

```diff
@@ -214,6 +214,11 @@ def ajax_str(self):
     def name(self):
         return u'%s %s' % (self.first_name, self.last_name)
 
+    def nonblank_name(self):
+        name = self.name()
+        if name.strip() == '': name = self.username
```

Comment:
I don't know how necessary this defensive logic really is; I copied it from `getTeacherNames`